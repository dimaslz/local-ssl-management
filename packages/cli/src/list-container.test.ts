import shell from "shelljs";

import listContainer from "./list-container";

vi.mock("chalk", async () => ({
  default: {
    green: vi.fn((v) => v),
    red: vi.fn((v) => v),
  },
}));

vi.mock("shelljs");

describe("List container", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe("success", () => {
    test("commmand found local-ssl-management container running", async () => {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (vi.spyOn(shell, "exec") as any).mockImplementation(() => {
        return {
          stdout:
            "XXXXXXXXXXXX | local-ssl-management | 0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp",
        };
      });

      listContainer();

      expect(shell.echo).toHaveBeenCalledTimes(2);
      expect(shell.echo).toMatchSnapshot();
    });
  });

  describe("failure", () => {
    test("some error happen", () => {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (vi.spyOn(shell, "exec") as any).mockImplementation(() => {
        return {
          stdout:
            "XXXXXXXXXXXX | something | 0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp",
        };
      });

      vi.spyOn(shell, "exit").mockImplementation(() => {
        throw new Error();
      });

      expect(() => {
        listContainer();
      }).toThrow();

      expect(shell.echo).toBeCalledTimes(1);
      expect(shell.exit).toBeCalledTimes(1);
    });
  });
});
