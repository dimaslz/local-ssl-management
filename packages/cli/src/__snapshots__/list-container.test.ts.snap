// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List container > success > commmand found local-ssl-management container running 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "
The local ssl proxy is running.

		ℹ️ The local ssl proxy is running. Keep it mind that it is important to the local domains that works through HTTPS.
",
    ],
    [
      "
[90m┌──────────────[39m[90m┬──────────────────────[39m[90m┬──────────────────────────────────────────┐[39m
[90m│[39m[31m container id [39m[90m│[39m[31m container image      [39m[90m│[39m[31m port                                     [39m[90m│[39m
[90m├──────────────[39m[90m┼──────────────────────[39m[90m┼──────────────────────────────────────────┤[39m
[90m│[39m XXXXXXXXXXXX [90m│[39m local-ssl-management [90m│[39m 0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp [90m│[39m
[90m└──────────────[39m[90m┴──────────────────────[39m[90m┴──────────────────────────────────────────┘[39m
",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
